### PR TITLE
optimize sync

### PR DESCRIPTION
**Pull request by JimmyShi22**
(1) Fake initializer for sync demo testing
(2) Optimize downloading logic and marco for more stable
(3) Move block decode logic from p2p callback to sync doWork
(4) Drop transactions when syncing
(5) Slow down sync loops

### CODE REVIEW

File: libsync/SyncMsgEngine.cpp [hunk 1]
```diff
@@ -115,8 +103,9 @@ bool SyncMsgEngine::interpret(SyncMsgPacket const& _packet)
 void SyncMsgEngine::onPeerStatus(SyncMsgPacket const& _packet)
 {
     shared_ptr<SyncPeerStatus> status = m_syncStatus->peerStatus(_packet.nodeId);
-    SyncPeerInfo info{_packet.nodeId, _packet.rlp()[0].toInt<int64_t>(),
-        _packet.rlp()[1].toHash<h256>(), _packet.rlp()[2].toHash<h256>()};
+    RLP const& rlps = _packet.rlp();
+    SyncPeerInfo info{
```
Comment (wheatli) [inline libsync/SyncMsgEngine.cpp]: check the itemCount of rlp before attempt to fetch them

Comment (wheatli) [reply libsync/SyncMsgEngine.cpp]: Would you plz to check all the parse codes when receive data from others

Comment (wheatli) [reply libsync/SyncMsgEngine.cpp]: Any mistake happens here will lead to DDos attack, ATTENTION PLZ

Comment (JimmyShi22) [reply libsync/SyncMsgEngine.cpp]: add count items logic and try catch logic

**Pull request merged**